### PR TITLE
ARROW-16895: [R] Fix cmake version detection

### PR DESCRIPTION
**Pull request by nealrichardson**
Discovered in https://github.com/apache/arrow/issues/13425

### CODE REVIEW

File: r/tools/nixlibs.R [hunk 1]
```diff
@@ -404,8 +404,10 @@ ensure_cmake <- function() {
   cmake
 }
 
-find_cmake <- function(paths, version_required = 3.10) {
+find_cmake <- function(paths, version_required = "3.10") {
   # Given a list of possible cmake paths, return the first one that exists and is new enough
+  # version_required should be a strong or packageVersion; numeric version
```
Comment (raulcd) [inline r/tools/nixlibs.R:409]:
```suggestion
  # version_required should be a string or packageVersion; numeric version
```

**Review by raulcd — COMMENTED** [review]
Minor typo

File: r/tools/nixlibs.R (hunk 1 above)
Comment (nealrichardson) [reply r/tools/nixlibs.R:409]: oof thanks

**Review by kou — APPROVED** [review]
+1

**Pull request merged**